third_party/amd/lib/TritonAMDGPUTransforms/Utility.h

```diff
@@ -0,0 +1,24 @@
+#ifndef TRITON_THIRD_PARTY_AMD_LIB_TRITONAMDGPUTRANSFORMS_UTILITY_H_
+#define TRITON_THIRD_PARTY_AMD_LIB_TRITONAMDGPUTRANSFORMS_UTILITY_H_
+
+#include "mlir/IR/Operation.h"
+#include "mlir/IR/Value.h"
+
+#include <functional>
+#include <limits>
+
+using namespace mlir;
+
+// DFS the def chain of 'value' starting from 'consumer' and will return the
```

Comment:
I'd suggest to rename `value` as `defValue` or something to be more clear. Also there will only ever be one path from `defValue` to `consumerOp` so DFS doesn't quite make sense to me--it's more forward walks IR from `defValue` to the `consumerOp` ..